packages/api/db/migration/20240111141747_create_animal_breed_table.js

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('animal_breed', (table) => {
+    table.increments('id').primary();
+    table.uuid('farm_id').references('farm_id').inTable('farm').defaultTo(null);
+    table.integer('type_id').references('id').inTable('animal_type').notNullable();
+    table.string('breed').defaultTo(null);
+    table.string('breed_key').defaultTo(null);
+    table.boolean('deleted').notNullable().defaultTo(false);
+    table.string('created_by_user_id').references('user_id').inTable('users');
+    table.string('updated_by_user_id').references('user_id').inTable('users');
+    table.dateTime('created_at').notNullable();
+    table.dateTime('updated_at').notNullable();
+    table.check('?? is not null or ?? is not null', ['breed', 'breed_key']);
+  });
+
+  // Add initial default breeds
+  const defaultBreedKeys = [
+    {
+      typeId: 1,
```

Comment:
I am concerned with using the index here versus searching for "CATTLE" key or something. Sometimes if a migration fails or if the entries are deleted, the table PK index count remains. 

If in the other PR the typeID is specifically added to make it the same as these type ids that is also fine. But right now it looks like they are added on a loop with the index taking whatever is next.

Reply:
Yes they have to be added in a loop using the auto-increment and not explicitly because we learned last time that not doing so breaks the table and causes the key unique violation error -- remember those fun times? 😄 https://github.com/LiteFarmOrg/LiteFarm/pull/2882

Reply:
Oh I remember I am always pro letting the auto increment do its thing! So searching for 'CATTLE' and 'farm_id = null' here would be best practice then here I think just in case something goes wrong with exact numbers.

Reply:
It's a good point, it did cross my mind but I wasn't sure. I'll do that 🙂